File: 2021/2021W40/preppindataW40.R
library(readr)
library(dplyr)
library(tidyr)
library(stringr)

final <- read_csv("C:/Data/PreppinData/Austin_Animal_Center_Outcomes.csv") %>%
  select('Animal Type', 'Outcome Type') %>%
  filter(`Animal Type` %in% c('Cat', 'Dog')) %>%
  mutate('Outcome Type' = if_else(str_detect(`Outcome Type`,'(Adoption|Return to Owner|Transfer)') & !is.na(`Outcome Type`),
                                   'Adopted, Returned to Owner or Transferred','Other')) %>%
  group_by_all() %>%
  summarise('Count' = n(), .groups='drop') %>%
  group_by(`Animal Type`) %>%
  mutate('Ratio' = round(`Count`*100/sum(`Count`),1)) %>%
  pivot_wider(id_cols='Animal Type', names_from = 'Outcome Type', values_from = 'Ratio', values_fn = max)

View(final)

File: 2021/2021W40/PreppindataW40.py
import pandas as pd
import re

animal = pd.read_csv("C:\\Data\\PreppinData\\Austin_Animal_Center_Outcomes.csv")[['Animal Type', 'Outcome Type']]
animal = animal[animal['Animal Type'].isin(['Cat', 'Dog'])].copy()
animal['Outcome Type'] = animal['Outcome Type'].apply(lambda x: 'Adopted, Returned to Owner or Transferred'
                                                      if re.match('(Adoption|Return to Owner|Transfer)',str(x)) else 'Other')
animal['RowCount'] = 1

final = animal.groupby(['Animal Type', 'Outcome Type']).agg({'RowCount':'sum'})
final['Ratio'] = final['RowCount'].groupby(level=0).apply(lambda x: round(x*100/float(x.sum()),1))
final = final.reset_index(drop=False)
final = final.pivot_table(index='Animal Type', columns='Outcome Type', values='Ratio', aggfunc=max).reset_index()
